hermes-common/src/main/java/pl/allegro/tech/hermes/common/message/converter/JsonToAvroConverter.java

```diff
@@ -0,0 +1,42 @@
+package pl.allegro.tech.hermes.common.message.converter;
+
+import org.apache.avro.AvroRuntimeException;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.io.BinaryEncoder;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.io.EncoderFactory;
+import org.apache.avro.io.JsonDecoder;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+public class JsonToAvroConverter {
+
+    public byte[] convert(byte[] data, String messageSchema) {
+        return convert(data, new Schema.Parser().parse(messageSchema));
```

Comment:
For every published message there is Schema object created from String. I think it can be overkill. Maybe we should use some cache for schemas.